hermes/probe/target/target.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove, GitHub: evanSpendlove.
+
+// Package target implements the Target struct used to hold all information
+// and state for a single probe run.
+package target
```

Comment:
I think that splitting this off now might brake the operations (create, read, and delete) that rely on this target. Maybe you could leave the splitting to a separate PR where you address those files too.

Reply:
I will double check those files now, but I'm splitting off into a separate package as when we ran Hermes on GCP, we ran into a cyclical import issue when target was in probe.go, so it needs to be moved elsewhere.

Reply:
Fair enough, then if you want to continue with the split in this PR, then please also include changes to the other files in this PR.

Reply:
I've included the changes in Create, Read and Delete in this PR.